### ISSUE TITLE
Atomically associate tls_private_key resources with a VM

### ISSUE THREAD

**Issue by dhbaird** [hashicorp/terraform#25012]
### Current Terraform Version

```
Terraform v0.12.24
```

### Use-cases

As an infrastructure creator, I want to ensure that every time I create a VM, that certain associated private material (tls_private_key, etc.) are also freshly generated and have never been used ever before, not even for a previous instantiation of the VM that was destroyed or replaced. Any data generated from these keys should also be re-generated, such as cloud-init user data templates that may incorporate this material to inject into the VM.

### Attempted Solutions

I've looked at using taint, the currently available lifecycle customizations, or triggers, but none of these options seem to do the right thing and require extra workflows to be built around terraform (e.g. "if I'm replacing or destroying a VM, don't forget to first taint the tls_private_keys"). These extra workflows introduce potential risks/bugs due to non-atomic operations only partially completing. For example: a tls_private_key gets tainted in anticipation of replacing a VM, but then the VM replace operation gets canceled or fails.

### Proposal

When creating, destroying, or replacing a VM, I would like for specifically designated tls_private_key resources to also be created, destroyed, or replaced. Furthermore, these operations should be atomic: if the VM couldn't be created, then the tls_private_key doesn't get created either (or rather, temporary state of its creation gets discarded). If a VM gets destroyed successfully, then so does its associated tls_private_key(s).

In other words, I would like the tls_private_key to act just like properties of the VM that are known upon its creation (things like IP address, id, etc.).

Thanks.

**Comment by dhbaird** [comment]
Closing as I figured how to achieve a close approximation of this through using modules.

**Comment by dhbaird** [comment]
P.S.- feel free to revive if someone thinks it's worth doing so.
